CONTRIBUTING.md

```diff
@@ -1,106 +1,186 @@
+2. Navigate to repository root directory
+3. Invoke `dotnet build eng\service.proj`
+
+## TO TEST:
+
+### Single Service from Command Line
+
+1. Open VS 2019 Command Prompt
+2. Navigate to service directory e.g. _"sdk\eventhub"_
+3. Invoke `dotnet test --filter TestCategory!=Live` _(Skips live tests)_
+4. or run test against **service.proj** in the repo root, passing the directory name of the specific service as a property. e.g. `dotnet test eng\service.proj /p:ServiceDirectory=eventhub --filter TestCategory!=Live`
+
+### Single Service from Visual Studio
+
+1. Build.
+2. Test Explorer window will get populated with tests. Select test and `Run` or `Debug`
+
+### All Client Services from Command Line
+
+1. Open VS 2019 Command Propmpt
+2. Navigate to repository root directory
+3. Invoke `dotnet test eng\service.proj --filter TestCategory!=Live`
+   <br/><br/>
+
+# Management Libraries
+
+## TO BUILD:
+
+### If you are building from Visual Studio, add a nuget feed source that points to < root >\tools\LocalNugetFeed directory
+
+1.  Open any solution, eg "SDKs\Compute\Microsoft.Azure.Management.Compute.sln"
+2.  Build solution from Visual Studio
+
+### Full Build from command line
+
+1.  Open VS 2019 command prompt
+2.  Navigate to repository root directory
+3.  Invoke `msbuild eng\mgmt.proj`
+
+### Single Service from Command Line
+
+1. Open VS 2019 command Prompt
+2. From the root directory
+3. Invoke `msbuild eng\mgmt.proj /p:scope=Compute`
+
+> _Build_ without any scope will build all management SDK's.
+
+### Create single nuget package
+
 In order to build one package and run it's test
-`msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
+`msbuild eng\mgmt.proj /t:CreateNugetPackage /p:scope=Compute`
 Nuget package will be created in root directory under \artifacts\packages\Debug (default configuration is Debug)
 
+## TO TEST:
+
+### Using Visual Studio:
+
+1. Build project in Visual Studio.
+2. **Test Explorer** window will get populated with tests. Select test and `Run` or `Debug`
+
+### Using the command line:
+
+Run e.g. `msbuild eng\mgmt.proj /t:"Runtests" /p:Scope=Compute`
+In the above example _RunTests_ will build and run tests for Compute only or you can use command line CLI
+`dotnet test Compute\Microsoft.Azure.Management.Compute\tests\Microsoft.Azure.Management.Tests.csproj`
+
 ### Non-Windows command line build
-Now you can use the same command on non-windows as above
-for e.g. on Ubuntu you can do something like below:
-`dotnet msbuild build.proj /t:Build /p:scope=Compute`
-`dotnet msbuild build.proj /t:RunTests /p:scope=Compute`
-`dotnet msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
+
+Now you can use the same command on non-windows as above for e.g. on Ubuntu you can do something like below:
+
+- `dotnet msbuild eng\mgmt.proj /t:Build /p:scope=Compute`
+- `dotnet msbuild eng\mgmt.proj /t:RunTests /p:scope=Compute`
+- `dotnet msbuild eng\mgmt.proj /t:CreateNugetPackage /p:scope=Compute`
```

Comment:
add the following
msbuild build.proj /t:Util /p:UtilityName=InstallPsModules
this will install the powershell modules that are needed for using generate.ps1

Reply:
This command does not work right now. Error: error MSB4057: The target "Util" does not exist in the project

Reply:
As we chatted I think we should hide that command completely and instead we should have a helper script under `eng` to handle this setup and then update all the `generate.ps1` to call that init script.

Reply:
@weshaggard changing generate.ps1 for every RP is a bigger change (at least I think that is little more involved as it will require all generate.ps1 to be updated and tested)
So unless we are planning to do that change right now, we need to have  a way for RPs to get the updated scripts documented, because if we do not do that, partners will start adding their own custom scripts in the repo and that will be even bigger workitem for a clean up.

@chidozieononiwu our partners have started using that target to get updated powershell scripts, and I am not able to reproduce the error you are mentioning.
The only way you end up getting that error if you are still using old tools.
let me know if you still see the issue if you do the following:
1. git clean -xdf
2. git pull upstream master
3. msbuild build.proj /t:Util /p:UtilityName=InstallPsModules

Reply:
While the change might be rather large, with a lot of files generate.ps1 files changed I don't think you need to test everyone just a few to verify it works. As it should be as simple as adding one line to the top of all generate.ps1.  If you don't think that is worth the effort right now might I suggest you go ahead and create a simple wrapper script in eng/mgmt that people can call instead of trying to have them remember and type this fairly complicated msbuild command?